OpenRA.Game/Network/OrderManager.cs

```diff
@@ -146,27 +146,24 @@ public void TickImmediate()
 						CheckSync(packet);
 					}
 					else if (frame == 0)
-						immediatePackets.Add((clientId, packet));
+					{
+						foreach (var o in packet.ToOrderList(World))
+						{
+							UnitOrders.ProcessOrder(this, World, clientId, o);
+
+							// A mod switch or other event has pulled the ground from beneath us
+							if (disposed)
+								return;
+						}
+					}
 					else
 					{
-						activeClients.Add(clientId);
-						pendingPackets.GetOrAdd(clientId).Enqueue(packet);
+						if (pendingPackets.TryGetValue(clientId, out var queue))
```

Comment:
```csharp
if (pendingPackets.TryGetValue(clientId, out var queue))
	queue.Enqueue(packet);
else
	Log.Write("debug", $"Received packet from disconnected client '{clientId}'");
```
?


Right now we log on normal clients and crash on disconnected!

Reply:
Doh, this is what lack of sleep does to a man.